subsys/bluetooth/mesh/settings.c

```diff
@@ -186,4 +192,18 @@ static void store_pending(struct k_work *work)
 void bt_mesh_settings_init(void)
 {
 	k_work_init_delayable(&pending_store, store_pending);
+	is_init = true;
+}
+
+void bt_mesh_settings_store_flush(void)
+{
+	if (is_init) {
+		k_work_flush_delayable(&pending_store, &work_sync);
+	}
+}
+
+void bt_mesh_settings_timeout_bypass(bool enable)
```

Comment:
I think I agree with @Laczen, this API is a bit troubling. If anything, the usage scenario of power down state storage should have functionality for doing the opposite of this, ie disabling timed storage altogether. If the application can detect power down, they can rely on a variant of `bt_mesh_settings_store_flush`, and won't need the timer at all. 
IMO, it would be better to introduce a kconfig entry like `CONFIG_BT_MESH_CONTINUOUS_STORE` or something that is enabled by default. If it's not enabled, the `STORE_TIMEOUT`, `SEQ_STORE_RATE` and `RPL_STORE_TIMEOUT` options can be hidden, and the entire timeout code in this module can be left out of the compilation. Then, all mesh store operations would have to be triggered by a call to `store_pending` from the application. The app can call this in its brown-out detection handler, or at any other time, if required.